java/ql/lib/ext/generated/org.apache.commons.net.model.yml

```diff
@@ -0,0 +1,1375 @@
+      - ["org.apache.commons.net", "DefaultSocketFactory", true, "DefaultSocketFactory", "(Proxy)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "MalformedServerReplyException", true, "MalformedServerReplyException", "(String)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "PrintCommandListener", true, "PrintCommandListener", "(PrintWriter)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "PrintCommandListener", true, "PrintCommandListener", "(PrintWriter,boolean)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "PrintCommandListener", true, "PrintCommandListener", "(PrintWriter,boolean,char)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "PrintCommandListener", true, "PrintCommandListener", "(PrintWriter,boolean,char,boolean)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "ProtocolCommandEvent", "(Object,String,String)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "ProtocolCommandEvent", "(Object,String,String)", "", "Argument[1]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "ProtocolCommandEvent", "(Object,String,String)", "", "Argument[2]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "ProtocolCommandEvent", "(Object,int,String)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "ProtocolCommandEvent", "(Object,int,String)", "", "Argument[2]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "getCommand", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandEvent", true, "getMessage", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
+      - ["org.apache.commons.net", "ProtocolCommandSupport", true, "ProtocolCommandSupport", "(Object)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "SocketClient", true, "connect", "(String)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "SocketClient", true, "connect", "(String,int)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["org.apache.commons.net", "SocketClient", true, "connect", "(String,int,InetAddress,int)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
```

Comment:
Good catch. Like almost always, I'm starting to have second thoughts about my own suggestions 😅. It's true that having the same model both as summary and sink can lead to duplicated paths, but also may be necessary for queries that don't use the sink (in which case the model only "exists" as a summary, which may be necessary for other paths).

In this case, though, these summaries don't seem to be necessary (it's hard to imagine what tainting a `SocketClient` even means), so it's probably a good idea to remove them. I hope you didn't review this PR line by line to catch this!

Another issue is manually modifying auto-generated files, because the changes would get overridden next time the generator is executed. So what we probably would need to do is applying this logic in the generator itself, if we decided to go this route.

About sources, it sounds like they should also supersede summaries, again to not replicate paths. But, same problem: what happens if the specific source isn't used in the analysis? Then we'd want the summary 😕.